third_party/move/move-compiler-v2/src/file_format_generator/peephole_optimizer.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module contains the peephole optimizer for the Move file format bytecode.
+//! Peephole optimizations assume that the bytecode is valid, and all user-facing
+//! error checks have already been performed.
+//! We currently do not perform peephole optimizations if inline spec blocks are present.
```

Comment:
This comment doesn't make sense here, as that decision is not made in this file.  More discussion of why it is a problem would be good, and probably moving the check to this file.

Reply:
Removed the comment.

Unfortunately, moving the check to this file is brittle: the only indication that there was an inline spec block in the bytecode instructions themselves is a `Nop` instruction. I don't think we should determine that indirectly through this brittle contract rather than directly available info at call site.